psi4/src/psi4/libmints/sieve.cc

```diff
@@ -307,6 +309,45 @@ void ERISieve::integrals() {
 #endif
 }
 
+void ERISieve::csam_integrals() {
+    function_sqrt_.resize((size_t)nbf_);
+    shell_pair_exchange_values_.resize(nshell_ * (size_t)nshell_);
+    std::fill(function_sqrt_.begin(), function_sqrt_.end(), 0.0);
+    std::fill(shell_pair_exchange_values_.begin(), shell_pair_exchange_values_.end(), 0.0);
+
+    IntegralFactory csamfactory(primary_, primary_, primary_, primary_);
+    std::shared_ptr<TwoBodyAOInt> eri = std::shared_ptr<TwoBodyAOInt>(csamfactory.eri());
+    const double *buffer = eri->buffer();
+
+    for (int P = 0; P < nshell_; P++) {
+        for (int Q = P; Q >= 0; Q--) {
+            int nP = primary_->shell(P).nfunction();
+            int nQ = primary_->shell(Q).nfunction();
+            int oP = primary_->shell(P).function_index();
+            int oQ = primary_->shell(Q).function_index();
+            eri->compute_shell(P, P, Q, Q);
+
+            // Computing Q_mu_mu (for denominator of Eq.9)
+            if (Q == P) {
+                int oP = primary_->shell(P).function_index();
+                for (int p = 0; p < nP; ++p) {
+                    function_sqrt_[oP + p] = std::sqrt(std::fabs(buffer[p * (nP * nP * nP + nP) + p * (nP * nP + 1)]));
```

Comment:
What's the Psi4 style: ```std::abs``` or ```std::fabs```?

IMHO it's a bit silly to stick to the ansi C name, since ```std::abs``` is polymorphic.

Reply:
Good point, `std::fabs` is probably present for historical reasons